pkg/controller/podautoscaler/metrics/rest_metrics_client_test.go

```diff
@@ -0,0 +1,274 @@
+
+				for i, metricPoint := range tc.reportedMetricPoints {
+					timestamp := fixedTimestamp.Add(time.Duration(metricPoint.timestamp) * time.Minute)
+					metric := cmapi.MetricValue{
+						DescribedObject: v1.ObjectReference{
+							Kind:       "Pod",
+							APIVersion: "v1",
+							Name:       fmt.Sprintf("%s-%d", podNamePrefix, i),
+						},
+						Value:      *resource.NewMilliQuantity(int64(metricPoint.level), resource.DecimalSI),
+						Timestamp:  metav1.Time{timestamp},
+						MetricName: tc.metricName,
+					}
+
+					metrics.Items = append(metrics.Items, metric)
+				}
+
+				return true, &metrics, nil
+			} else {
+				name := getForAction.GetName()
+				mapper := api.Registry.RESTMapper()
+				assert.NotNil(t, tc.singleObject, "should have only requested a single-object metric when we asked for metrics for a single object")
+				gk := schema.FromAPIVersionAndKind(tc.singleObject.APIVersion, tc.singleObject.Kind).GroupKind()
+				mapping, err := mapper.RESTMapping(gk)
+				if err != nil {
+					return true, nil, fmt.Errorf("unable to get mapping for %s: %v", gk.String(), err)
+				}
+				groupResource := schema.GroupResource{Group: mapping.GroupVersionKind.Group, Resource: mapping.Resource}
+
+				assert.Equal(t, groupResource.String(), getForAction.GetResource().Resource, "should have requested metrics for the resource matching the GroupKind passed in")
+				assert.Equal(t, tc.singleObject.Name, name, "should have requested metrics for the object matching the name passed in")
+				metricPoint := tc.reportedMetricPoints[0]
+				timestamp := fixedTimestamp.Add(time.Duration(metricPoint.timestamp) * time.Minute)
+
+				metrics := &cmapi.MetricValueList{
+					Items: []cmapi.MetricValue{
+						{
+							DescribedObject: v1.ObjectReference{
+								Kind:       tc.singleObject.Kind,
+								APIVersion: tc.singleObject.APIVersion,
+								Name:       tc.singleObject.Name,
+							},
+							Timestamp:  metav1.Time{timestamp},
+							MetricName: tc.metricName,
+							Value:      *resource.NewMilliQuantity(int64(metricPoint.level), resource.DecimalSI),
+						},
+					},
+				}
+
+				return true, metrics, nil
+			}
+		})
+	}
+
+	return fakeMetricsClient, fakeCMClient
+}
+
+func (tc *restClientTestCase) verifyResults(t *testing.T, metrics PodMetricsInfo, timestamp time.Time, err error) {
+	if tc.desiredError != nil {
+		assert.Error(t, err, "there should be an error retrieving the metrics")
+		assert.Contains(t, fmt.Sprintf("%v", err), fmt.Sprintf("%v", tc.desiredError), "the error message should be eas expected")
+		return
+	}
+	assert.NoError(t, err, "there should be no error retrieving the metrics")
+	assert.NotNil(t, metrics, "there should be metrics returned")
+
+	assert.Equal(t, tc.desiredMetricValues, metrics, "the metrics values should be as expected")
+
+	targetTimestamp := fixedTimestamp.Add(time.Duration(tc.targetTimestamp) * time.Minute)
```

Comment:
you have to cast to a `time.Duration` before you can multiply it by a `time.Duration` (which `time.Minute` is).  It's the same way in the legacy rest client test.